gc/gc-base/src/main/java/org/projectnessie/gc/base/DistributedIdentifyContents.java

```diff
@@ -0,0 +1,392 @@
+  }
+
+  /**
+   * Gets the expired contents per content id by walking all the live and dead references in a
+   * distributed way using spark and checking the contents against the live bloom filter results.
+   *
+   * @param liveContentsBloomFilterMap live contents bloom filter per content id.
+   * @param references list of all the references to walk (live and dead)
+   * @return {@link IdentifiedResult} object.
+   */
+  public IdentifiedResult getIdentifiedResults(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, List<Reference> references) {
+
+    List<IdentifiedResult> results =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, getTaskCount(references))
+            .map(reference -> computeExpiredContents(liveContentsBloomFilterMap, reference))
+            .collect();
+    IdentifiedResult identifiedResult = new IdentifiedResult();
+    results.forEach(
+        result -> identifiedResult.getContentValues().putAll(result.getContentValues()));
+    return identifiedResult;
+  }
+
+  protected Map<String, ContentBloomFilter> computeLiveContents(
+      Instant cutOffTimestamp,
+      Reference reference,
+      Instant droppedRefTime,
+      Long totalCommitsInDefaultReference) {
+    try (NessieApiV1 api = getApi(config)) {
+      boolean isRefDroppedAfterCutoffTimeStamp =
+          droppedRefTime == null || droppedRefTime.compareTo(cutOffTimestamp) >= 0;
+      return walkLiveCommitsInReference(
+          api,
+          reference,
+          isRefDroppedAfterCutoffTimeStamp,
+          commitMeta ->
+              // If the commit time is newer than (think: greater than or equal to) cutoff-time,
+              // then commit is live.
+              commitMeta.getCommitTime().compareTo(cutOffTimestamp) >= 0,
+          totalCommitsInDefaultReference);
+    }
+  }
+
+  protected @NotNull IdentifiedResult computeExpiredContents(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, Reference reference) {
+    try (NessieApiV1 api = getApi(config)) {
+      return walkAllCommitsInReference(api, reference, liveContentsBloomFilterMap);
+    }
+  }
+
+  private Instant getCutoffTimeForRef(
+      Reference reference,
+      Map<String, Instant> cutOffTimestampPerRef,
+      Instant defaultCutOffTimestamp,
+      Instant deadReferenceCutOffTimeStamp,
+      Map<String, Instant> droppedRefTimeMap) {
+    if (droppedRefTimeMap.containsKey(reference.getName() + ":" + reference.getHash())
+        && deadReferenceCutOffTimeStamp != null) {
+      return deadReferenceCutOffTimeStamp;
+    }
+    return cutOffTimestampPerRef.getOrDefault(reference.getName(), defaultCutOffTimestamp);
+  }
+
+  private int getTaskCount(List<Reference> references) {
+    int taskCount =
+        Integer.parseInt(
+            config.getOrDefault(
+                GCConfigConstants.CONF_NESSIE_GC_SPARK_TASK_COUNT,
+                String.valueOf(references.size())));
+    if (taskCount <= 0) {
```

Comment:
if the config's invalid and the implementation silently behaves in an unexpected way, that's surprising for users. better error out early.

Reply:
moved all conf validation to beginning